### PR TITLE
Add `cargo zaplib serve`

### LINKED ISSUE #16: Move server from Python 2 to `cargo zaplib`
Python 2 is old and unsupported; we've had multiple people complain about this.

### PR DESCRIPTION
**Pull request by janpaul123**
Instead of an old Python 2 script. Closes #16.

I needed to serve files for the cross-browser CI testing, and I didn’t
want to fiddle with Python 2, so I figured I’d just do this now.

Will publish this crate when merged.

See individual commits!

### CODE REVIEW

File: zaplib/cargo-zaplib/src/cmd.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
+use log::info;
+
+use std::process::{exit, Command};
+
+use clap::{App, AppSettings, Arg};
+
+pub(crate) fn cmd() {
```
Comment (janpaul123) [inline zaplib/cargo-zaplib/src/cmd.rs:7]: Most of this code is moved from main.rs (see comment there).

**Comment by stevekrouse** [comment]
I checked out this branch and tried running `zaplib serve` and it just said `command not found: zaplib`. Did I miss the step for how to install `zaplib`?

File: zaplib/docs/src/contributing.md [hunk 2]
```diff
@@ -6,16 +6,13 @@ There are 2 types of tests available. The browser test suite is more extensive,
...
+  * Have local server running: `zaplib serve`
   * Navigate to `http://localhost:3000/zaplib/test_suite/` and click `Run All Tests` button
 * Running jest tests
   * `cd zaplib/web && yarn run jest`
 
 ## Updating the documentation
 
 * Build the website and watch for changes: `zaplib/scripts/watch_website_dev.sh` 
-* Run local server on port 4848: `website_dev/server.sh` 
+* Run local server: `zaplib serve website_dev/ --port 4848`
```
Comment (stevekrouse) [inline zaplib/docs/src/contributing.md:17]: I'd expect it without the final slash command and to have a default port (4848 or whatever) included 

```suggestion
* Run local server: `zaplib serve website_dev` 
```

File: zaplib/cargo-zaplib/src/cmd.rs (hunk 1 above)
Comment (stevekrouse) [reply zaplib/cargo-zaplib/src/cmd.rs:3]: Makes sense

**Comment by stevekrouse** [comment]
Removing the python 2 dependency is huge, thanks JP!

**Comment by stevekrouse** [comment]
Question - is it possible to bundle this command into cargo watch or something? Would be nice if you didn't have to run two commands

File: zaplib/docs/src/contributing.md (hunk 2 above)
Comment (janpaul123) [reply zaplib/docs/src/contributing.md:17]: Actually let me change it a bit more even

File: zaplib/cargo-zaplib/src/build.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+pub(crate) struct BuildOpts {
+    pub(crate) release: bool,
+    pub(crate) use_simd128: bool,
+    pub(crate) all_targets: bool,
+    pub(crate) workspace: bool,
+    pub(crate) package: String,
+    pub(crate) features: String,
+}
+
+pub(crate) fn build(opts: BuildOpts) {
```
[the remark this answers is not in the document]
Comment (janpaul123) [reply zaplib/cargo-zaplib/src/build.rs:15]: I just made a second commit to split out this build.rs file also, for consistency (see individual commits).

**Comment by janpaul123** [comment]
> I checked out this branch and tried running zaplib serve and it just said command not found: zaplib. Did I miss the step for how to install zaplib?

That won't work until I publish this change to crates.io. To run the local version of this library, use `cargo run -p cargo-zaplib -- serve`

> Question - is it possible to bundle this command into cargo watch or something? Would be nice if you didn't have to run two commands

Yeah good point. Added a ticket: #59.

File: zaplib/docs/src/contributing.md (hunk 2 above)
Comment (janpaul123) [reply zaplib/docs/src/contributing.md:17]: Maybe it's nice to keep the 4848 in case you already have `zaplib serve` running, so the ports don't conflict?

**Review by stevekrouse — APPROVED** [review]

**Pull request merged**